File: code/java/src/main/java/composite/Song.java
package composite;

class Song implements PlaylistComponent {
    private String title;

    Song(String title) {
        this.title = title;
    }

    @Override
    public void playSong() {
        // Perform song-specific operation, such as playing
        System.out.println("Playing song: " + title);
    }
}

File: code/typescript/composite/Song.ts
import { IPlaylistComponent } from "./IPlaylistComponent";

export class Song implements IPlaylistComponent {
    private title: string;

    constructor(title: string) {
        this.title = title;
    }

    playSong(): void {
        console.log(`Playing song: ${this.title}`);
    }
}
